controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1547,3 +1625,146 @@ func (r *NnfWorkflowReconciler) getContainerVolumes(ctx context.Context, workflo
+		if err != nil {
+			return nil, err
+		}
+
+		// Check to see if we've already made an allocation
+		for _, alloc := range pm.Spec.Allocations {
+			if alloc.Requester.UID == workflow.UID {
+				return nil, nil
+			}
+		}
+
+		// Add a port allocation request to the manager for the number of ports specified by the
+		// container profile
+		pm.Spec.Allocations = append(pm.Spec.Allocations, nnfv1alpha1.NnfPortManagerAllocationSpec{
+			Requester: corev1.ObjectReference{
+				Name:      workflow.Name,
+				Namespace: workflow.Namespace,
+				Kind:      reflect.TypeOf(dwsv1alpha2.Workflow{}).Name(),
+				UID:       workflow.UID,
+			},
+			Count: int(profile.Data.NumPorts),
+		})
+
+		if err := r.Update(ctx, pm); err != nil {
+			if !apierrors.IsConflict(err) {
+				return nil, err
+			}
+			return Requeue("update port manager allocation"), nil
+		}
+
+		r.Log.Info("Ports Requested", "numPorts", profile.Data.NumPorts)
+	}
+
+	return nil, nil
+}
+
+// Ensure that the default NnfPortManager has assigned the appropriate number of requested ports
+func (r *NnfWorkflowReconciler) checkContainerPorts(ctx context.Context, workflow *dwsv1alpha2.Workflow, index int) (*result, error) {
+
+	profile, err := getContainerProfile(ctx, r.Client, workflow, index)
+	if err != nil {
+		return nil, err
+	}
+
+	// Nothing to do here if ports are not requested
+	r.Log.Info("Checking for requested ports", "numPorts", profile.Data.NumPorts)
+	if profile.Data.NumPorts > 0 {
+		pm, err := getContainerPortManager(ctx, r.Client)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, alloc := range pm.Status.Allocations {
+			if alloc.Requester != nil && alloc.Requester.UID == workflow.UID {
+				if alloc.Status == nnfv1alpha1.NnfPortManagerAllocationStatusInUse && len(alloc.Ports) == int(profile.Data.NumPorts) {
+					// Add workflow env var for the ports
+					name, val := getContainerPortsEnvVar(alloc.Ports)
+					workflow.Status.Env[name] = val
+					return nil, nil // done
+				} else if alloc.Status == nnfv1alpha1.NnfPortManagerAllocationStatusInvalidConfiguration {
+					return nil, nnfv1alpha1.NewWorkflowError("error requesting ports for container workflow: Invalid NnfPortManager configuration").WithFatal()
+				} else if alloc.Status == nnfv1alpha1.NnfPortManagerAllocationStatusInsufficientResources {
+					return nil, nnfv1alpha1.NewWorkflowError("error requesting ports for container workflow: InsufficientResources").WithFatal()
+				}
+			}
+		}
+
+		return Requeue("NnfPortManager allocation not ready").after(2 * time.Second).withObject(pm), nil
+	}
+
+	return nil, nil
+}
+
+// Retrieve the default NnfPortManager for user containers. Allow a client to be passed in as this
+// is meant to be used by reconcilers or container helpers.
+func getContainerPortManager(ctx context.Context, cl client.Client) (*nnfv1alpha1.NnfPortManager, error) {
+	pm := &nnfv1alpha1.NnfPortManager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nnf-port-manager",
```

Comment:
This name and namespace should be sent in via env vars:
https://github.com/NearNodeFlash/nnf-sos/blob/master/config/manager/manager.yaml#L110
and
https://github.com/NearNodeFlash/nnf-sos/blob/master/controllers/suite_test.go#L74

Reply:
Done